### PR TITLE
Implement Copy Permalink as Markdown share provider

### PR DESCRIPTION
**Pull request by joyceerhl**
For https://github.com/microsoft/vscode/issues/176316

### CODE REVIEW

File: src/issues/shareProviders.ts [hunk 1]
```diff
@@ -194,9 +202,49 @@ export class GitHubPermalinkShareProvider extends AbstractShareProvider implemen
...
+				return selectedText;
+			}
+
+			const wordRange = document.getWordRangeAtPosition(item.selection.start);
+			if (wordRange) {
+				return document.getText(wordRange);
+			}
+		}
+
+		return fileName;
```
Comment (joyceerhl) [inline src/issues/shareProviders.ts:241]: This implementation should address https://github.com/microsoft/vscode-pull-request-github/issues/4663 once we inline Share (which requires async context menus in Electron)

**Comment by joyceerhl** [comment]
@alexr00 it's hidden by default but surfaces in the command palette when you enable `"workbench.experimental.share.enabled": true` and also next to the command center when you enable `"window.commandCenter": true`.

**Comment by alexr00** [comment]
@joyceerhl the 0.66.0 release has happened. It is ok to merge this when you're ready!

**Comment by joyceerhl** [comment]
Thank you @alexr00!

**Pull request merged**